internal/db/id/field.go

```diff
@@ -12,69 +12,160 @@ package id
 
 import (
 	"context"
+	"errors"
+	"strconv"
 
-	"github.com/sourcenetwork/immutable"
+	"github.com/sourcenetwork/corekv"
 
 	"github.com/sourcenetwork/defradb/client"
-	"github.com/sourcenetwork/defradb/client/request"
-	"github.com/sourcenetwork/defradb/datastore"
 	"github.com/sourcenetwork/defradb/internal/db/sequence"
+	"github.com/sourcenetwork/defradb/internal/db/txnctx"
 	"github.com/sourcenetwork/defradb/internal/keys"
 )
 
-// SetFieldIDs sets the field IDs hosted on the given collections, mutating the input set.
-func SetFieldIDs(ctx context.Context, txn datastore.Txn, definitions []client.CollectionDefinition) error {
-	schemasByName := map[string]client.SchemaDescription{}
-	for _, def := range definitions {
-		schemasByName[def.Schema.Name] = def.Schema
+// GetShortFieldID returns the local, shortened, internal, field id, which is used
+// only in locations where using the full CID would be a waste of storage space.
+func GetShortFieldID(
+	ctx context.Context,
+	collectionShortID uint32,
+	fieldID string,
+) (uint32, error) {
+	// This concatenation is temporary, soon we can just use the field CID
+	uniqueKey := strconv.Itoa(int(collectionShortID)) + ":" + fieldID
+
+	cache := getFieldShortIDCache(ctx)
+	shortID, ok := cache[uniqueKey]
+	if ok {
+		return shortID, nil
+	}
+
+	// If we miss the cache, load the entire collection's worth - it is almost always
+	// going to be more efficient than loading the field short-ids one by one, and we'll
+	// usually want most of them.
```

Comment:
question: Have you thought about caching these globally instead of per call? I'm thinking even caching the collection descriptions would improve performance.

Reply:
That would get fairly complicated RE transactions.  Long term we'll probably want to do that, but right now I do not think it is worth the dev-effort.  Doing it this way in the medium term should not waste much time (actually implementing the cache was a handful of minutes here).